transcom/mymove
# INT-B-19872 - Allow Office Users to View Uploaded Supporting Documents

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19872)

## Summary

This branch adds a "Supporting Documents" tab to the navigation when you are viewing a move. This tab is available to all office users. Once the user clicks on it, they are taken to a document viewer where they can view all Supporting Documents that the customer has uploaded. They can also download these files, and view a file list on the left side.

It also adds the needed functionality to the backend, mostly adding "additionalDocuments" to the payload when you issue a GET request to a the `move/{locator}` endpoint.


### How to test

First, make sure your .envrc has the flag "FEATURE_FLAG_MANAGE_SUPPORTING_DOCS" set to true (it should by default).

1. As a customer, access/create any move, and submit it to the SC, you should now see the "Upload Additional Documents" button on your move dashboard. Upload some documents here.
2. As an SC, access the move and verify that you see the "Supporting Documents" tab. Click on it.
3. Verify that you can see your newly uploaded docs in this viewer, and that you can see all of them in the file list.
4. Repeat steps 2-3 for each office role.
5. Edit the feature flag in .envrc to "FALSE", stop server, re-run `direnv allow`, re-run server
6. Verify that the Supporting Documents tab does *not* show up for any office role.

## Screenshots
![Screenshot 2024-06-27 at 1 26 23 PM](https://github.com/transcom/mymove/assets/146959895/ffc8b214-d8ac-4955-9e79-f2b4c36ecbf0)
![Screenshot 2024-06-27 at 1 26 38 PM](https://github.com/transcom/mymove/assets/146959895/41e6a8a8-b10e-402e-b800-5f71b12d8d69)
![Screenshot 2024-06-27 at 1 26 52 PM](https://github.com/transcom/mymove/assets/146959895/1eafc7a8-6219-41f7-aea0-b6b023e46a63)
